src/common/types/blob.cpp

```diff
@@ -43,27 +43,27 @@ uint64_t Blob::getBlobSize(const ku_string_t& blob) {
     return blobSize;
 }
 
-void Blob::fromString(ku_string_t& str, uint8_t* resultBuffer) {
+uint64_t Blob::fromString(const char* str, uint64_t length, uint8_t* resultBuffer) {
```

Comment:
I guess you can have `Blob::fromString(const char*, uint64_t, uint8_t*)` and `Blob::fromString(const ku_string_t&, uint8_t*)`.

We have the same design for string functions